src/detect-geoip.c

```diff
@@ -84,27 +84,85 @@ void DetectGeoipRegister(void)
  * \internal
  * \brief This function is used to initialize the geolocation MaxMind engine
  *
- * \retval NULL if the engine couldn't be initialized
- * \retval (GeoIP *) to the geolocation engine
+ * \retval FALSE if the engine couldn't be initialized
  */
-static GeoIP *InitGeolocationEngine(void)
+static int InitGeolocationEngine(DetectGeoipData *geoipdata)
 {
-    return GeoIP_new(GEOIP_MEMORY_CACHE);
+    const char *filename = NULL;
+
+    /* Get location and name of GeoIP2 database from YAML conf */
+    (void)ConfGet("geoip-database", &filename);
+
+    if (filename == NULL) {
```

Comment:
would it make sense to use a default location in this case?

Reply:
The new MaxMind library, when passed a NULL filename, will return a file open error and terminate.  This behavior is different from the legacy libGeoIP library which would search a default path for a DB file before giving up.

Do you want to include all the code baggage of locating a default path and DB file across Linux/Windows platforms, or just log the error message if a valid DB file and path is not given in suricata.yaml?

Reply:
After further reflection, I propose leaving the code "as-is" whereby it logs an appropriate warning message and disables the geoip rule option when the GeoIP2 database is not specified in _suricata.yaml_.  The _libmaxminddb_ library will not use a default path when NULL is provided for the DB name.  There are at least two different DB names possible for the Country database depending on whether the user has the free GeoLite2 edition or the paid subscription GeoIP2 edition.  The user is responsible for configuring the DB download anyway, so the user may as well specify the location of the DB to Suricata via the _suricata.yaml_ entry.